src/miniscript/astelem.rs

```diff
@@ -848,70 +848,79 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Terminal<Pk, Ctx> {
     ///
     /// This number does not include the witness script itself, so 1 needs
     /// to be added to the final result.
-    pub fn max_satisfaction_witness_elements(&self) -> usize {
+    pub fn max_satisfaction_witness_elements(&self) -> Option<usize> {
         match *self {
-            Terminal::PkK(..) => 1,
-            Terminal::PkH(..) => 2,
-            Terminal::After(..) | Terminal::Older(..) => 0,
+            Terminal::PkK(..) => Some(1),
+            Terminal::PkH(..) => Some(2),
+            Terminal::After(..) | Terminal::Older(..) => Some(0),
             Terminal::Sha256(..)
             | Terminal::Hash256(..)
             | Terminal::Ripemd160(..)
-            | Terminal::Hash160(..) => 1,
-            Terminal::True => 0,
-            Terminal::False => 0,
+            | Terminal::Hash160(..) => Some(1),
+            Terminal::True => Some(0),
+            Terminal::False => Some(0),
```

Comment:
This was a bug in previous code too. I think this should be None.

Reply:
Yeah definitely, it is in my "track costs" branch and in Sipa's implementation.